infra-as-code/bicep/orchestration/hubPeeredSpoke/hubPeeredSpoke.bicep

```diff
@@ -0,0 +1,206 @@
+@maxLength(10)
+param parTopLevelManagementGroupPrefix string = 'alz'
+
+@description('Subscription Id to the Virtual Network Hub object. DEFAULTS TO empty')
+param parPeeredVnetSubscriptionId string = ''
+
+@description('Array of Tags to be applied to all resources in module. Default: empty array')
+param parTags object = {}
+
+@description('Set Parameter to true to Opt-out of deployment telemetry DEFAULTS TO = false')
+param parTelemetryOptOut bool = true
+
+// Subscription Module Parameters
+@description('The Management Group Id to place the subscription in. DEFAULTS TO empty')
+@allowed([
+  'landingZonesCorp'
+  'identity'
+  'connectivity'
+  ''
+])
+param parPeeredVnetSubscriptionMGPlacement string = ''
+
+// Resource Group Module Parameters
+@description('Name of Resource Group to be created to contain spoke networking resources like the virtual network.  Default: {parTopLevelManagementGroupPrefix}-{parLocation}-spoke-networking')
+param parResourceGroupNameForSpokeNetworking string = '${parTopLevelManagementGroupPrefix}-${parLocation}-spoke-networking'
+
+// Spoke Networking Module Parameters
+@description('Existing DDoS Protection plan to utilize. Default: Empty string')
+param parDdosProtectionPlanId string = ''
+
+@description('The Name of the Spoke Virtual Network. Default: vnet-spoke')
+param parSpokeNetworkName string = 'vnet-spoke'
+
+@description('CIDR for Spoke Network. Default: 10.11.0.0/16')
+param parSpokeNetworkAddressPrefix string = '10.11.0.0/16'
+
+@description('Array of DNS Server IP addresses for VNet. Default: Empty Array')
+param parDNSServerIPArray array = []
+
+@description('IP Address where network traffic should route to. Default: Empty string')
+param parNextHopIPAddress string = ''
+
+@description('Switch which allows BGP Route Propogation to be disabled on the route table')
+param parBGPRoutePrapogation bool = false
+
+@description('Name of Route table to create for the default route of Hub. Default: rtb-spoke-to-hub')
+param parSpoketoHubRouteTableName string = 'rtb-spoke-to-hub'
+
+// Peering Modules Parameters
+@description('Virtual Network ID of Hub Virtual Network, or Azure Virtuel WAN hub ID. No default')
+param parHubVirtualNetworkID string
+
+@description('Switch to enable/disable forwarded Traffic from outside spoke network. Default = false')
+param parAllowSpokeForwardedTraffic bool = false
+
+@description('Switch to enable/disable VPN Gateway for the hub network peering. Default = false')
+param parAllowHubVPNGatewayTransit bool = false
+
+// **Variables**
+// Customer Usage Attribution Id
+var varCuaid = '8ea6f19a-d698-4c00-9afb-5c92d4766fd2'
+
+// Orchestration Module Variables
+var varTargetManagementGroupID = (parPeeredVnetSubscriptionMGPlacement =='connectivity' ? '${parTopLevelManagementGroupPrefix}-platform-connectivity' : (parPeeredVnetSubscriptionMGPlacement =='identity' ? '${parTopLevelManagementGroupPrefix}-platform-identity' : '${parTopLevelManagementGroupPrefix}-landingzones-corp'))
```

Comment:
Think this will need amending as we are remove the "allowed" list for this parameter?

Reply:
Updated in newest commit